stock_picking_package_preparation_line/models/stock.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    Copyright 2015 Lorenzo Battistini - Agile Business Group
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, api
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    def get_packs(self):
+        packs = self.env['stock.picking.package.preparation']
+        pack_line_model = self.env['stock.picking.package.preparation.line']
+        pack_lines = pack_line_model.search([
+            ('move_id', '=', self.id),
```

Comment:
since the method seems to always be called within a `for move in self` loop, why not move this loop in get_packs?

You just have to change the search domain to ('move_id', 'in', self.ids)